app/views/papers/_tablist.html.erb

```diff
@@ -1,7 +1,14 @@
 <% paper = reference.paper %>
+<% comments_count = reference.comments.map {|c| c.self_and_descendants.length}.inject(&:+) %>
 <ul class="nav nav-pills" role="tablist">
   <% %i{abstract details comments}.each do |tab| %>
-    <li><%= link_to tab.capitalize, '', "data-detail-id" => "#{tab}-#{tab_id_for(paper)}", class: "small paper-detail-tab" %></li>
+    <li>
+      <%= link_to "#{tab.capitalize}" +
```

Comment:
It would only do a string substitution (`String#gsub`) if it matched on "Comments." I was going to note that it was a _tiny_ performance hit since you're checking the string for comments each time, but then the above code already does this. To be honest, there are so few tabs with the same settings that it might be better just to write them all out at this point.